### ISSUE TITLE
Cannot run output of > dotnet ef migrations script -i

### ISSUE THREAD

**Issue by aureole82** [dotnet/efcore#21496]
1) Model:  
    ```cs
    public class Ticket
    {
        public int Id { get; set; }
        public string Reference { get; set; }
        public string ExternalId { get; set; }
    }
    ```
    `> dotnet ef migrations add -p .\WebApplication\WebApplication.csproj "Initial"`

2) Change: _public string External~~Id~~Number { get; set; }_
`> dotnet ef migrations add -p .\WebApplication\WebApplication.csproj "ChangeToExternalNumber"`

3) `> dotnet ef migrations script -i -p .\WebApplication\WebApplication.csproj -o .\migrate.sql
[WebApplication.zip](https://github.com/dotnet/efcore/files/4865518/WebApplication.zip)
`

Running migrate.sql will work only once, the 2nd time:
> Msg 207, Level 16, State 1, Line 33
> **Invalid column name 'ExternalId'.**
> 
> Completion time: 2020-07-02T17:46:03.7876814+02:00

---
_(Entity Framework Core .NET Command-line Tools 3.1.5)_
[WebApplication.zip](https://github.com/dotnet/efcore/files/4865523/WebApplication.zip)

**Comment by ajcvickers** [comment]
Here's the SQL we generate:

```
IF OBJECT_ID(N'[__EFMigrationsHistory]') IS NULL
BEGIN
    CREATE TABLE [__EFMigrationsHistory] (
        [MigrationId] nvarchar(150) NOT NULL,
        [ProductVersion] nvarchar(32) NOT NULL,
        CONSTRAINT [PK___EFMigrationsHistory] PRIMARY KEY ([MigrationId])
    );
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200706202846_Initial')
BEGIN
    CREATE TABLE [Ticket] (
        [Id] int NOT NULL IDENTITY,
        [Reference] nvarchar(max) NULL,
        [ExternalId] nvarchar(max) NULL,
        CONSTRAINT [PK_Ticket] PRIMARY KEY ([Id])
    );
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200706202846_Initial')
BEGIN
    INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
    VALUES (N'20200706202846_Initial', N'3.1.3');
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200706203045_ChangeToExternalNumber')
BEGIN
    DECLARE @var0 sysname;
    SELECT @var0 = [d].[name]
    FROM [sys].[default_constraints] [d]
    INNER JOIN [sys].[columns] [c] ON [d].[parent_column_id] = [c].[column_id] AND [d].[parent_object_id] = [c].[object_id]
    WHERE ([d].[parent_object_id] = OBJECT_ID(N'[Ticket]') AND [c].[name] = N'ExternalId');
    IF @var0 IS NOT NULL EXEC(N'ALTER TABLE [Ticket] DROP CONSTRAINT [' + @var0 + '];');
    ALTER TABLE [Ticket] DROP COLUMN [ExternalId];
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200706203045_ChangeToExternalNumber')
BEGIN
    ALTER TABLE [Ticket] ADD [ExternalNumber] nvarchar(max) NULL;
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200706203045_ChangeToExternalNumber')
BEGIN
    INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
    VALUES (N'20200706203045_ChangeToExternalNumber', N'3.1.3');
END;

GO
```

**Comment by aureole82** [comment]
Sorry, but that's the migration you've create.
`PS C:\Users\user\Projects\WebApplication> dotnet ef migrations script -i -p .\WebApplication\WebApplication.csproj -o .\migrate.sql`

```sql
IF OBJECT_ID(N'[__EFMigrationsHistory]') IS NULL
BEGIN
    CREATE TABLE [__EFMigrationsHistory] (
        [MigrationId] nvarchar(150) NOT NULL,
        [ProductVersion] nvarchar(32) NOT NULL,
        CONSTRAINT [PK___EFMigrationsHistory] PRIMARY KEY ([MigrationId])
    );
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702160931_Initial')
BEGIN
    IF SCHEMA_ID(N'ticket') IS NULL EXEC(N'CREATE SCHEMA [ticket];');
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702160931_Initial')
BEGIN
    CREATE TABLE [ticket].[Tickets] (
        [Id] int NOT NULL IDENTITY,
        [Reference] nvarchar(450) NULL,
        [ExternalId] nvarchar(450) NULL,
        CONSTRAINT [PK_Tickets] PRIMARY KEY ([Id])
    );
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702160931_Initial')
BEGIN
    CREATE UNIQUE INDEX [IX_Tickets_ExternalId] ON [ticket].[Tickets] ([ExternalId]) WHERE [ExternalId] IS NOT NULL;
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702160931_Initial')
BEGIN
    CREATE INDEX [IX_Tickets_Reference] ON [ticket].[Tickets] ([Reference]);
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702160931_Initial')
BEGIN
    INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
    VALUES (N'20200702160931_Initial', N'3.1.5');
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702161106_ChangeToExternalNumber')
BEGIN
    DROP INDEX [IX_Tickets_ExternalId] ON [ticket].[Tickets];
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702161106_ChangeToExternalNumber')
BEGIN
    DECLARE @var0 sysname;
    SELECT @var0 = [d].[name]
    FROM [sys].[default_constraints] [d]
    INNER JOIN [sys].[columns] [c] ON [d].[parent_column_id] = [c].[column_id] AND [d].[parent_object_id] = [c].[object_id]
    WHERE ([d].[parent_object_id] = OBJECT_ID(N'[ticket].[Tickets]') AND [c].[name] = N'ExternalId');
    IF @var0 IS NOT NULL EXEC(N'ALTER TABLE [ticket].[Tickets] DROP CONSTRAINT [' + @var0 + '];');
    ALTER TABLE [ticket].[Tickets] DROP COLUMN [ExternalId];
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702161106_ChangeToExternalNumber')
BEGIN
    ALTER TABLE [ticket].[Tickets] ADD [ExternalNumber] nvarchar(450) NULL;
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702161106_ChangeToExternalNumber')
BEGIN
    CREATE UNIQUE INDEX [IX_Tickets_ExternalNumber] ON [ticket].[Tickets] ([ExternalNumber]) WHERE [ExternalNumber] IS NOT NULL;
END;

GO

IF NOT EXISTS(SELECT * FROM [__EFMigrationsHistory] WHERE [MigrationId] = N'20200702161106_ChangeToExternalNumber')
BEGIN
    INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
    VALUES (N'20200702161106_ChangeToExternalNumber', N'3.1.5');
END;

GO
```

**Comment by ajcvickers** [comment]
Duplicate of #12911 which we should consider for 5.0.
